crates/polars-ops/src/series/ops/replace.rs

```diff
@@ -33,7 +33,20 @@ pub fn replace(
         return Ok(default);
     }
 
-    let old = old.strict_cast(s.dtype())?;
+    let old = match (s.dtype(), old.dtype()) {
+        (s_dt, old_dt) if s_dt == old_dt => old.clone(),
```

Comment:
Isn't this logic already in `strict_cast` / `cast`? I think it should be implemented there, rather than before every cast everywhere else.

Reply:
@ritchie46 , I don't think it is right now. Would it be sensible to create a fastpath in `strict_cast` / `cast`?

Reply:
It is in `Series::cast`. I think we should dispatch `strict_cast` via that. Done that in #13255.

So this cane be rebased and this branch can be removed.